### PR TITLE
Milestone for builtin libraries/modules

### PR DESCRIPTION
**Pull request by TypeMonkey**
Added mile stone progress code, including editions for code and example programs that were chosen for testing.

### CODE REVIEW

File: ast.ts [hunk 1]
```diff
@@ -92,20 +163,21 @@ export type Expr<A> =
   | { a?: A; tag: "uniop"; op: UniOp; expr: Expr<A> }
   | { a?: A; tag: "builtin1"; name: string; arg: Expr<A> }
   | { a?: A; tag: "builtin2"; name: string; left: Expr<A>; right: Expr<A> }
-  | { a?: A; tag: "call"; name: string; arguments: Array<Expr<A>> }
+  | { a?: A; tag: "call"; name: string; arguments: Array<Expr<A>>, callSite?: CallSite }
   // ASSIGNABLE EXPRS
   | { a?: A; tag: "id"; name: string }
   | { a?: A; tag: "lookup"; obj: Expr<A>; field: string }
   // END ASSIGNABLE EXPRS
```
Comment (jpolitz) [inline ast.ts:170]: Why are a bunch of thigns re-ordered here? This is difficult to merge.

File: compiler.ts [hunk 2]
```diff
@@ -1,142 +1,121 @@
...
+{tag: "globalvar", index: number} |
+{tag: "module", originalName: string} 
 
-export const emptyEnv: GlobalEnv = {
-  globals: new Map(),
-  classes: new Map(),
-  locals: new Set(),
-  offset: 0,
-};
+const INTANCTIATE = "1nstanciate";
```
Comment (jpolitz) [inline compiler.ts:23]: Instantiate?

File: compiler.ts [hunk 3]
```diff
@@ -1,142 +1,121 @@
...
-function envLookup(env: GlobalEnv, name: string): number {
-  if (!env.globals.has(name)) {
-    console.log("Could not find " + name + " in ", env);
-    throw new Error("Could not find name " + name);
-  }
-  return env.globals.get(name) * 4; // 4-byte values
-}
 
-function codeGenStmt(stmt: Stmt<Type>, env: GlobalEnv): Array<string> {
+function codeGenStmt(stmt: Stmt<Type>, idens: Array<Map<string, IdenType>>, labels: LabeledComps): Array<string> {
```
Comment (jpolitz) [inline compiler.ts:69]: Why remove env? Should these extra arguments become a part of env?

File: compiler.ts [hunk 4]
```diff
@@ -167,9 +146,14 @@ function codeGenStmt(stmt: Stmt<Type>, env: GlobalEnv): Array<string> {
       return [...objStmts, `(i32.add (i32.const ${offset * 4}))`, ...valStmts, `(i32.store)`];
     default:
       unhandledTag(stmt);
+      */
   }
 }
 
+function codeGenExpr(expr: Expr, idens: Array<Map<string, IdenType>>, labels: LabeledComps): string {
```
Comment (jpolitz) [inline compiler.ts:153]: ?

**Review by jpolitz — COMMENTED** [review]
Leaving this as a comment so you can confirm – is this intended to merge?

**Comment by TypeMonkey** [comment]
For documentation purposes: I'll be closing this pull request. I can seem to make PRs for the same branch, which contains my submission for March 11.

**Pull request closed without merging**